File: types/healthkit.go
package types

import (
	"time"

	"github.com/google/uuid"
)

type BaseHKData struct {
	ID             uuid.UUID `json:"id"`
	Type           string    `json:"type"`
	Value          string    `json:"value"`
	StartTimestamp string    `json:"start_timestamp"`
	EndTimestamp   string    `json:"end_timestamp"`
	TimezoneID     string    `json:"timezone_id"`
}

type BaseHealthKit struct {
	StartTime time.Time    `json:"start_time"`
	EndTime   time.Time    `json:"end_time"`
	Data      []BaseHKData `json:"data"`
}

File: client/react-demo/src/types/healthKit.ts
export interface HKData {
  id: string;
  type: string;
  value: string;
  startTimestamp: string;
  endTimestamp: string;
  timezoneID: string;
}

export interface HealthKit {
  id: string;
  startTime: Date;
  endTime: Date;
  data: HKData[];
}
